Runtime/Shaders/ShaderHelper.cs

```diff
@@ -0,0 +1,299 @@
+                    out var glossTexture);
+
+            // If there are no textures, don't bake anything, slider values can control everything:
+            if (!texturesExist ||
+                glossTexture == null ||
+                rgbTexture == null) {
+                destMat.SetTexture(channelName, null);
+
+                if (channelName == BaseMeshSync._MetallicGlossMap) {
+                    destMat.DisableKeyword(BaseMeshSync._METALLICGLOSSMAP);
+                    destMat.DisableKeyword(BaseMeshSync._METALLICSPECGLOSSMAP);
+                }
+
+                return;
+            }
+
+            var shader = LoadShader(SHADER_NAME_SMOOTHNESS_INTO_ALPHA);
+
+            shader.SetTexture(SHADER_CONST_SMOOTHNESS, glossTexture);
+            shader.SetTexture(SHADER_CONST_RGB, rgbTexture);
+
+            var texture = shader.RenderToTexture(destMat.GetTexture(channelName));
+
+            if (channelName == BaseMeshSync._MetallicGlossMap) {
+                destMat.EnableKeyword(BaseMeshSync._METALLICGLOSSMAP);
+                destMat.EnableKeyword(BaseMeshSync._METALLICSPECGLOSSMAP);
+            }
+            else if (smoothnessChannel == 0) {
+                destMat.DisableKeyword(BaseMeshSync._METALLICGLOSSMAP);
+                destMat.DisableKeyword(BaseMeshSync._METALLICSPECGLOSSMAP);
+            }
+
+            destMat.SetTexture(channelName, texture);
+        }
+#endif
+
+        public static void BakeMaps(Material destMat,
+            List<TextureHolder> textureHolders,
+            List<MaterialPropertyData> materialProperties) {
+#if AT_USE_HDRP
+            BakeMaskMap(destMat, textureHolders, materialProperties);
+#else
+            BakeSmoothness(destMat, textureHolders, materialProperties);
+#endif
+        }
+
+        //        private static Texture TextureFromRenderTexture(RenderTexture renderTarget, string path) {
+        //            // TODO:
+        //            // Saving the render texture out to the assetlibrary is really slow, use the render texture and save it later instead:
+        //#if THIS_IS_TOO_SLOW_AT_RUNTIME
+        //            var texture = new Texture2D(renderTarget.width, renderTarget.height, UnityEngine.TextureFormat.RGBA32, true);
+        //            RenderTexture.active = renderTarget;
+        //            texture.ReadPixels(new Rect(0, 0, renderTarget.width, renderTarget.height), 0, 0);
+        //            texture.Apply();
+
+        //            TextureData.WriteToFile(path, texture.EncodeToPNG());
+
+        //            if (string.IsNullOrEmpty(AssetDatabase.AssetPathToGUID(path))) { 
+        //                AssetDatabase.ImportAsset(path);
+        //            }
+        //            var savedTexture = AssetDatabase.LoadAssetAtPath<Texture2D>(path);
+
+        //            return savedTexture;
+        //#endif
+        //            return renderTarget;
+        //        }
+
+        #region Internals
```

Comment:
It would make things a bit cleaner/easier to understand if seaprate the baking logic and this `Internals` region functionality into a separate classes/files

Reply:
Done